### ISSUE TITLE
WinAppDriver.exe is not opening after installing it

### ISSUE THREAD

**Issue by Aya-Akl** [Microsoft/WinAppDriver#179]
Hey

I have an issue that when trying to open WinAppDriver.exe, a console appears to be opening then nothing is displayed.

When I try running the calculator test after running appium in command line, the following error is displayed:

**C:/Ruby23/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.3.0/lib/selenium/webdriver/remote/http/common.rb:90:in `create_response': unexpected response, code=404, content-type="text/plain" (Selenium::WebDriver::Error::WebDriverError)
The URL '/session' did not map to a valid resource**

My configurations are as follows:

```
def caps
    {
        platformName: "Windows", deviceName: "WindowsPC", app: "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App"
    }
end

def setup
    $CalculatorSession = Selenium::WebDriver.for(:remote, :url => "http://127.0.0.1:4723/", :desired_capabilities => caps)
end
```

My PC is running on Windows 10 Enterprise and I enabled the Developer Mode.

Is there something I am missing here?!

Thanks!

**Comment by khouzam** [comment]
@Aya-Akl 

Can you open a command-prompt and run WinAppDriver manually, this will at least show what the error message is without having the window close on you.

**Comment by Aya-Akl** [comment]
@khouzam 
It worked when I opened it manually using the command-prompt and no errors were displayed. The script ran successfully now. Thank you so much!

But the window is still closing on me when I open the .exe file itself. Maybe there is something wrong with my Windows OS?

**Comment by yodurr** [comment]
I can't think of why WinAppDriver would get closed because of how you opened it.  
What if you right click the .exe and select Run As Administrator.  Does that make a difference?

**Comment by yodurr** [comment]
Hi @Aya-Akl did this problem go away or are you still running into it?

**Comment by Aya-Akl** [comment]
@yodurr yes it did go away. I never encountered it again but honestly I don't know why. I tried ro Run as administrator as you told me before but it was not working then I tried it again and it just worked.

Thank you so much for your help.

**Comment by vijay44** [comment]
Hi @yodurr,

I think I'm facing the same issue which was discussing above but using **java** and **jUnit**. 

- Developer mode is enabled on windows 10.
-  I have tried to run winappdriver.exe manually for the default port, but couldn't run since appium is already using port no 4723.

When I run the calculator example using java I'm getting below error;

> org.openqa.selenium.UnsupportedCommandException: The URL '/session' did not map to a valid resource
> Command duration or timeout: 155 milliseconds
> Build info: version: '2.53.0', revision: '35ae25b1534ae328c771e0856c93e187490ca824', time: '2016-03-15 10:43:46'
> System info: host: 'LKCONVSELVARA01', ip: '10.83.68.58', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_131'
> Driver info: io.appium.java_client.ios.IOSDriver


My code is as below
```
private static IOSDriver CalculatorSession = null;
   private static WebElement CalculatorResult = null;

@BeforeClass
    public static void setup() {
        try {
            DesiredCapabilities capabilities = new DesiredCapabilities();
            capabilities.setCapability("app", "C:\\Windows\\System32\\calc.exe");
             CalculatorSession = new IOSDriver(new URL("http://127.0.0.1:4723"), capabilities);
            CalculatorSession.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);

            CalculatorSession.findElementByName("Clear").click();
            CalculatorSession.findElementByName("Seven").click();
            CalculatorResult = CalculatorSession.findElementByName("Display is  7 ");
            Assert.assertNotNull(CalculatorResult);

        }catch(Exception e){
            e.printStackTrace();
        } finally {
        }
    }

```
```
1. Am doing anything wrong here? or do I need to do any additional configuration  other than in the video?
2. Do I need to run the winappdriver.exe explicitly? If so it should be a different than 4723 (default)? because appium is running on port '4723'

```

Thanks in advance.

**Comment by janiekbuysrogge** [comment]
@vijay44 : Did you try with the Appium URL pointing to http://127.0.0.1:4723**/wd/hub** ?

**Comment by yodurr** [comment]
@janiekbuysrogge is correct Appium requires /wd/hub on the URL if you're pointing the test at Appium (which is then proxying to WinAppDriver)

**Comment by vijay44** [comment]
Thanks @yodurr & @janiekbuysrogge. Working fine with IOS driver

**Comment by timotiusmargo** [comment]
Closing as the issue posted has been solved.
